membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -21,36 +21,46 @@ class AttributesFromZuora(implicit val executionContext: ExecutionContext) exten
   def getAttributes(identityId: String,
                     identityIdToAccountIds: String => Future[String \/ QueryResponse],
                     subscriptionsForAccountId: AccountId => SubPlanReads[AnyPlan] => Future[Disjunction[String, List[Subscription[AnyPlan]]]],
+                    accountSummaryForAccountId: AccountId => Future[\/[String, AccountSummary]],
+                    paymentMethodForPaymentMethodId: PaymentMethodId => Future[\/[String, PaymentMethodResponse]],
                     dynamoAttributeService: AttributeService,
                     forDate: LocalDate = LocalDate.now()): Future[(String, Option[Attributes])] = {
 
+
     def twoWeekExpiry = forDate.toDateTimeAtStartOfDay.plusDays(14)
 
     val attributesFromDynamo: Future[Option[DynamoAttributes]] = dynamoAttributeService.get(identityId) map { attributes =>
       dynamoAttributeService.ttlAgeCheck(attributes, identityId, forDate)
       attributes
     }
 
-    val zuoraAttributesDisjunction: DisjunctionT[Future, String, Option[ZuoraAttributes]] = for {
+    val zuoraAttributesDisjunction: DisjunctionT[Future, String, Future[Option[ZuoraAttributes]]] = for {
       accounts <- EitherT[Future, String, QueryResponse](withTimer(s"ZuoraAccountIdsFromIdentityId", zuoraAccountsQuery(identityId, identityIdToAccountIds), identityId))
       accountIds = queryToAccountIds(accounts)
-      subscriptions <- EitherT[Future, String, List[Subscription[AnyPlan]]](
-        if(accountIds.nonEmpty) withTimer(s"ZuoraGetSubscriptions", getSubscriptions(accountIds, identityId, subscriptionsForAccountId), identityId)
+      accountSummaries <- EitherT[Future, String, List[AccountSummary]](
+        if(accountIds.nonEmpty) withTimer(s"ZuoraGetAccountSummaries", getAccountSummaries(accountIds, identityId, accountSummaryForAccountId), identityId)
```

Comment:
Is it possible to get the info we need without making an additional REST call? Can we select the fields that we care about as part of the query that we run on line 39 instead?

Reply:
Happy to investigate this as a follow up, rather than slowing down this PR though, as I know we're keen to get this released